### PR TITLE
Add Google Chat support

### LINKED ISSUE #102: Add Google Chat support
**Motivation**

Google Chat(former Google Hangout Chat) adoption is increasing these days.

**Feature**

Send notification like Slack and Mattermost

**Alternatives**

**Additional context**

May I add this feature? 
I would like to contribute.

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

> /area config

> /area outputs

> /area tests

> /area helm

**What this PR does / why we need it**:

[Google Chat](https://chat.google.com/welcome) is a wonderful workspace app like Slack, Mattermost, etc.

image

**Which issue(s) this PR fixes**:

Fixes #102 

**Special notes for your reviewer**:

The API description of the Google Chat API is described here https://developers.google.com/hangouts/chat/how-tos/webhooks.

### CODE REVIEW

**Comment by Issif** [comment]
Awesome, I'll review it asap. Thank you.

File: README.md [hunk 1]
```diff
@@ -338,8 +344,11 @@ The *env vars* "match" field names in *yaml file with this structure (**take car
 * **GCP_PUBSUB_PROJECTID**: The GCP Project ID containing the Pub/Sub Topic
 * **GCP_PUBSUB_TOPIC**: The name of the Pub/Sub topic
 * **GCP_PUBSUB_MINIMUMPRIORITY**: minimum priority of event for using this output, order is `emergency|alert|critical|error|warning|notice|informational|debug or "" (default)`
+* **GOOGLECHAT_WEBHOOKURL** : Google Chat URL (ex: https://chat.googleapis.com/v1/spaces/XXXXXX/YYYYYY), if not `empty`, Google Chat output is *enabled*
+* **ROCKETCHAT_OUTPUTFORMAT** : `all` (default), `text` (only text is displayed in Google Chat)
+* **GOOGLECHAT_MINIMUNPRIORITY** : minimum priority of event for using this output, order is `emergency|alert|critical|error|warning|notice|informational|debug or "" (default)`
```
Comment (Issif) [inline README.md:350]:
```suggestion
* **GOOGLECHAT_MINIMUMPRIORITY** : minimum priority of event for using this output, order is `emergency|alert|critical|error|warning|notice|informational|debug or "" (default)`
```

File: README.md [hunk 2]
```diff
@@ -235,6 +236,12 @@ gcp:
     projectid: "" # The GCP Project ID containing the Pub/Sub Topic
     topic: "" # The name of the Pub/Sub topic
   # minimumpriority: "debug" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
+
+googleChat:
```
Comment (KeisukeYamashita) [inline README.md:240]: Thank you for your quick reply.
Should this field be `googlechat` not `googleChat`?

File: outputs/slack.go [hunk 3]
```diff
@@ -119,6 +120,8 @@ func newSlackPayload(falcopayload types.FalcoPayload, config *types.Configuratio
 	// 	iconURL = config.Slack.Icon
 	// }
 
+	log.Println(messageText)
```
Comment (Issif) [inline outputs/slack.go]:
```suggestion
```

File: outputs/slack.go [hunk 4]
```diff
@@ -41,6 +41,7 @@ func newSlackPayload(falcopayload types.FalcoPayload, config *types.Configuratio
 	var field slackAttachmentField
 
 	if config.Slack.OutputFormat == All || config.Slack.OutputFormat == Fields || config.Slack.OutputFormat == "" {
+		log.Print("not text!")
```
Comment (Issif) [inline outputs/slack.go]:
```suggestion
```

**Review by Issif — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Issif) [reply README.md:240]: `googlechat` for respecting a global pattern

Comment (KeisukeYamashita) [reply README.md:240]: Thank you, I will fix this now 👍

Comment (KeisukeYamashita) [reply README.md]: @Issif Thank you for your reviews. I will this in https://github.com/falcosecurity/falcosidekick/pull/107/commits/0d31fe33b713bacbe673fc2cf1ec371508fa6b3c 🙇

**Review by Issif — DISMISSED** [review]

**Comment by Issif** [comment]
@KeisukeYamashita just a little conflict to fix and it's all good 👍

**Comment by KeisukeYamashita** [comment]
Sorry, I accidentally merge the master branch... PTAL when you have time. Thank you in advance 🙇

**Review by Issif — APPROVED** [review]

**Review by Issif — APPROVED** [review]

**Review by Issif — APPROVED** [review]

**Review by Issif — DISMISSED** [review]

**Comment by leogr** [comment]
Hi @KeisukeYamashita 

Thank you for your contribution :+1: 

Since we do not allow merge commits, could you please fix that by removing the [Merge branch 'master' into add-google-chat-support](https://github.com/falcosecurity/falcosidekick/pull/107/commits/cea9684e7d99ffa45bfd596752c5b96b07e09f68) commit and then perform a `git rebase -i master`?

We will be happy to merge this PR once that has been fixed :rocket: 

If you need any support on that, please let me know.

**Comment by KeisukeYamashita** [comment]
@leogr  Sorry, I removed the merge commit 🙇  Thank you for your kind notice and guide 🙇 

image

**Review by leogr — APPROVED** [review]
All right, LGTM now :+1: 

Thank you @KeisukeYamashita :clap:

**Pull request merged**